### PR TITLE
Add earthly multi-platform build process

### PR DESCRIPTION
**Pull request by Cliftonz**
Added earthly build but to build properly I needed to NOT use go workspaces, have a global mod file, and move internal packages to the lib folder.

### CODE REVIEW

**Comment by jasekiw** [comment]
Is the limit to 3 users going to be an issue for us? 
image

**Comment by Cliftonz** [comment]
> Is the limit to 3 users going to be an issue for us? 
> image
> 

Not sure but If we hit an issue I can self host on my server which would increase it to 5 users

**Comment by jasekiw** [comment]
@Cliftonz What are the benefits of using earthly that will outweigh these restrictions?

**Comment by Cliftonz** [comment]
> @Cliftonz What are the benefits of using earthly that will outweigh these restrictions?

Earthly natively supports consistent build processes containerized approach which ensures consistent and reproducible builds across all environments. Earthly's support for parallel execution, caching, and seamless CI/CD integration will speed up our pipelines and reduce redundancy. Unlike Makefile, Earthly offers modern syntax and advanced features tailored for complex, multi-platform projects, making it ideal for our containerized and cloud-native workflows. This shift will improve our build consistency, performance, and scalability compared to makefile, which would gain substantial complexity over time.

File: test/load-test/simpleloadtest.js [hunk 1]
```diff
@@ -10,9 +10,18 @@ export let options = {
 };
 
 export default function () {
```
Comment (EdWrld) [inline test/load-test/simpleloadtest.js:12]: I think more stressful testing is required here

**Comment by EdWrld** [comment]
In the future, if you could split out multiple features into separate prs or mention all the major things you've done in the description that'll make it easier to review and not miss anything.

**Comment by Cliftonz** [comment]
> In the future, if you could split out multiple features into separate prs or mention all the major things you've done in the description, that'll make it easier to review and not miss anything.

You're right, this was my bad. Let me fix this.

**Pull request merged**